packages/contracts/foundry_test/TimelockOperations.t.sol

```diff
@@ -0,0 +1,312 @@
+            address(activePool),
+            abi.encodeCall(activePool.sweepToken, (address(mockToken), amountToSweep)),
+            7 days + 1
+        );
+        vm.stopPrank();
+
+        // confirm balances
+        address feeRecipientAddress = activePool.feeRecipientAddress();
+
+        assertEq(mockToken.balanceOf(address(activePool)), amountInActivePool - amountToSweep);
+        assertEq(mockToken.balanceOf(address(feeRecipientAddress)), amountToSweep);
+    }
+
+    function test_onlyHighsecHasAdmin() public {
+        // Dummy minter
+        address minter = _utils.getNextSpecialAddress();
+        assertFalse(authority.doesUserHaveRole(minter, 1)); // No role assigned yet
+
+        // Attempt to perform adming function from lowSec Timelock
+        vm.startPrank(techOps);
+        _scheduleTimelockTx(
+            lowSecTimelock,
+            address(authority),
+            abi.encodeCall(
+                authority.setUserRole,
+                (minter, 1, true) // minter
+            ),
+            2 days + 1
+        ); // Can schedule it but executing should revert
+        vm.warp(block.timestamp + 2 days + 1);
+        vm.expectRevert("TimelockController: underlying transaction reverted");
+        _executeTimelockTx(
+            lowSecTimelock,
+            address(authority),
+            abi.encodeCall(
+                authority.setUserRole,
+                (minter, 1, true) // minter
+            )
+        );
+        vm.stopPrank();
+
+        // Attempt to schedule from highSec timelock with lowsec account (permissions test)
+        vm.startPrank(techOps);
+        vm.expectRevert();
+        _scheduleTimelockTx(
+            highSecTimelock,
+            address(authority),
+            abi.encodeCall(
+                authority.setUserRole,
+                (minter, 1, true) // minter
+            ),
+            7 days + 1
+        );
+        vm.stopPrank();
+
+        // Attemt to operate admin functions from previous governance
+        vm.prank(defaultGovernance);
+        vm.expectRevert();
+        authority.setUserRole(minter, 1, true);
+
+        // High sec user can use highSec timelock to perform admin operations (extensible minting, for instance)
+        vm.startPrank(ecosystem);
+        _scheduleAndExecuteTimelockTx(
+            highSecTimelock,
+            address(authority),
+            abi.encodeCall(
+                authority.setUserRole,
+                (minter, 1, true) // minter
+            ),
+            7 days + 1
+        );
+        vm.stopPrank();
+        assertTrue(authority.doesUserHaveRole(minter, 1));
+    }
+
+    /// @dev Helper function to grant all Governor setter capabilities to a specific role
+    /// @dev Assumes default governance still has ownerships and is pranked
+    function _grantAllGovernorCapabilitiesToRole(uint8 role) internal {
+        // List of all setter function signatures in Governor contract
+        bytes4[] memory funcSigs = new bytes4[](9);
```

Comment:
the `funcSigs` has a length of `9` but following assignment seems miss index `3` and `5` ?

Reply:
Good catch